src/invidious/videos/parser.cr

```diff
@@ -6,7 +6,7 @@ require "json"
 #
 # TODO: "compactRadioRenderer" (Mix) and
 # TODO: Use a proper struct/class instead of a hacky JSON object
-def parse_related_video(related : JSON::Any) : Hash(String, JSON::Any)?
+def parse_related_video(related : JSON::Any, published) : Hash(String, JSON::Any)?
```

Comment:
You can probably just make `published` into an optional argument here. Also, I'd prefer if you can just initialize the JSON::Any below. 

The related video data struct can then look something like 
` "published"        => JSON::Any.new(published || "0"),`